### ISSUE TITLE
Difficulty in using submodule as developing folder

### ISSUE THREAD

**Issue by popabczhang** [CityScope/CSL_Shanghai_LivingLine#1]
After some research and tests, I find quite hard to use submodule as developing folder. Please refer to this article for the difficulty: 

[https://blog.github.com/2016-02-01-working-with-submodules/](https://blog.github.com/2016-02-01-working-with-submodules/)

Quotes: 
"Remember that Git doesn’t download submodule contents by default. If you’re adding a submodule to an existing project, make sure anyone that works on the project knows they need to run commands like git submodule update and git clone --recursive to ensure they get everything. "

"Submodules require you to carefully balance consistency and convenience. "

"Remember that collaborators won’t automatically see updates to submodules—if you update a submodule, you may need to remind your colleagues to run git submodule update or they will likely see odd behavior."

"Managing dynamic, rapidly evolving or heavily co-dependent repositories with submodules can quickly become frustrating. "

To sum up, it's very hard to have every party to develop and coordinate inside a submodule. Considering the ever approaching deadline for LivingLine CityScope table, I think a temporary solution, for now, is to keep the submodule in the project as a reference, develop the unity project in a separate folder, and cite the original repo properly. 

P.S.: I really feel folk is the way to go. Unfortunately, we can not folk a repo in the same Org. We should find a way work around it.

**Comment by RELNO** [comment]
Thanks @popabczhang. While I'll wait for @yasushisakai & @agrignard  input on this, and considering the short timeline, why don't we converge to work on the [already made] branch? 
[https://github.com/CityScope/CS_CSharpScope_Template/tree/CSL_Shangahi_LivingLine](url) 

I know you and Tongji wants a dedicated repo, but simply cloning with no reference to source is kinda missing the whole versioning idea. Citation is really secondary to all this, it's about have a  trackable development process.

**Comment by popabczhang** [comment]
Some thoughts:

1.
This way, the history is still trackable practically because you can still
see all the commits before and after the clone. You just need to
combine/track them "manually". State and cite clearly in the wiki and
README is enough in my opinion.

2.
The fact that we can not folk to the same Org actually indicates that each
CSL shall have their own Org.

3.
I don't think it's a good idea to develop a project under template repo.

4.
The project can wait, but not indefinitely. Let's set up a deadline for
deciding what's the way to go -- by end of this week (Saturday 11:59pm
EST).

**Comment by RELNO** [comment]
> 1. This way, the history is still trackable practically because you can still
see all the commits before and after the clone. You just need to
combine/track them "manually". State and cite clearly in the wiki and
README is enough in my opinion.

Not sure how manual combine works, especially for Unity, any idea? 

> 2. The fact that we can not folk to the same Org actually indicates that each
CSL shall have their own Org.

That's a bigger question, probably for a discussion AFTER we're done with the Summit. IMO, if much bigger projects can keep themselves in one org, we should too. 

> 3. I don't think it's a good idea to develop a project under template repo
> 4.The project can wait, but not indefinitely. Let's set up a deadline for
deciding what's the way to go -- by end of this week (Saturday 11:59pm
EST).

There is no reason to wait at all-- dev can take place in the branch for now and any decision could be applied at any future point. 

I totally understand the pressure and the urgency, but we should try and set some long term goals for CS so we won't end up with dozens of one-off projects as before.

**Comment by yasushisakai** [comment]
ordered by recommendation.

## Option 1: use living line branch
I would use the CS_CSharpScope's living line branch, update the submodule less frequently (since we don't have interdependencies, you can actually forget about it, submodules are nothing than pointers to commits.)

When doing that use the issues thoroughly so that @RELNO will be 'pushed' to resolve it. This way the Living Line project gains one more head(brain) count, and the repo will maximize the change to be **better** from experiences doing a real project, making it robust for future deployments. Think about it, just adding comments and deciphering the current code base is already a tremendous contribution. If you split we loose it. Ariel should be a good dictator, with power but responsibility. (well nothing stops the living line people to take over the master if you like, its the maintainer's role to tag or make a release making it easy to roll back, branch and restart from there.)

## Option 2: Fork to individual account and do PR when needed.
As from @RELNO, argument 2 is I guess out of scope, but as an alternative, I don't see a reason not forking to an individual account and then do a pull request. (in the case, the implications are that it's less attractive to collaborate with the maintainer) but it's traceable and clean. or it's just a git organization after all, we are all learning, so might be good to experiment the idea of making an organization.

## Option 3: Copy and paste (or clone) then commit
Generally people who copies and then commits will be less credible, since it is unclear who did what. Plus, I suppose this repo will be a mixture several things, making it harder to see the isolated history of a specific feature. I do understand @popabczhang 's 3rd argument, (so I will post an issue on the github.io repo) yet disagree, because again, the tools are not made by themselves but nurtured by the city projects. Chicken and egg.
